api/src/controllers/report.js

```diff
@@ -0,0 +1,40 @@
+const auth = require('../auth');
+const ctx = require('../services/data-context');
+const serverUtils = require('../server-utils');
+const { Report, Qualifier } = require('@medic/cht-datasource');
+
+const getReport = () => ctx.bind(Report.v1.get);
+const getReportIds = () => ctx.bind(Report.v1.getIdsPage);
+
+const checkUserPermissions = async (req) => {
+  const userCtx = await auth.getUserCtx(req);
+  if (!auth.isOnlineOnly(userCtx) || !auth.hasAllPermissions(userCtx, 'can_view_contacts')) {
+    return Promise.reject({ code: 403, message: 'Insufficient privileges' });
+  }
+};
+
+module.exports = {
+  v1: {
+    get: serverUtils.doOrError(async (req, res) => {
+      await checkUserPermissions(req);
+      const { uuid } = req.params;
+      const report = await getReport()(Qualifier.byUuid(uuid));
+
+      if (!report) {
+        return serverUtils.error({ status: 404, message: 'Report not found' }, req, res);
+      }
+
+      return res.json(report);
+    }),
+    getIds: serverUtils.doOrError(async (req, res) => {
+      await checkUserPermissions(req);
+
+      const qualifier = Qualifier.byFreetext(req.query.freetext);
```

Comment:
nope. it returns a `400 - Bad request` error because `freetext` is required whereas `limit` is set to a default of 10000.
![image](https://github.com/user-attachments/assets/04ee5176-962b-4c6e-a71a-8a29775cf0fa)